imagebuildah/stage_executor.go

```
@@ -1782,18 +1789,29 @@ func (s *StageExecutor) generateCacheKey(ctx context.Context, currNode *parser.N
 
 // cacheImageReference is internal function which generates ImageReference from Named repo sources
 // and a tag.
-func cacheImageReferences(repos []reference.Named, cachekey string) ([]types.ImageReference, error) {
+func cacheImageReferences(repos []types.ImageReference, cachekey string) ([]types.ImageReference, error) {
 	var result []types.ImageReference
 	for _, repo := range repos {
-		tagged, err := reference.WithTag(repo, cachekey)
-		if err != nil {
-			return nil, fmt.Errorf("failed generating tagged reference for %q: %w", repo, err)
-		}
-		dest, err := imagedocker.NewReference(tagged)
-		if err != nil {
-			return nil, fmt.Errorf("failed generating docker reference for %q: %w", tagged, err)
+		if repo.Transport().Name() == imagedocker.Transport.Name() {
+			ref := repo.DockerReference()
+			tagged, err := reference.WithTag(ref, cachekey)
+			if err != nil {
+				return nil, fmt.Errorf("failed generating tagged reference for %q: %w", repo, err)
+			}
+			dest, err := imagedocker.NewReference(tagged)
+			if err != nil {
+				return nil, fmt.Errorf("failed generating docker reference for %q: %w", tagged, err)
+			}
+			result = append(result, dest)
+		} else if repo.Transport().Name() == imagedirectory.Transport.Name() {
+			dirPath := repo.StringWithinTransport()
+			tagged := filepath.Join(dirPath, cachekey)
+			dest, err := imagedirectory.NewReference(tagged)
+			if err != nil {
+				return nil, fmt.Errorf("failed generating directory reference for %q: %w", tagged, err)
+			}
+			result = append(result, dest)
 		}
```

`else`… this silently drops unsupported input without telling the user?!